src/features/environments/components/EnvironmentDropdown.tsx

```diff
@@ -100,8 +100,10 @@ export const EnvironmentDropdown = ({
             }
           >
             <StyledIconButton
-              onClick={e => onCreateNewEnvironmentTab(e, namespace)}
-              disabled={!canCreate}
+              onClick={e =>
+                canCreate && onCreateNewEnvironmentTab(e, namespace)
+              }
+              aria-disabled={!canCreate}
```

Comment:
This was a legit issue caught by the warnings. If you reveal lines 95-99 just above, you'll see that the create button is wrapped in a tooltip that is supposed to explain what the button does or why it is disabled. However, if you set the HTML `disabled` attribute on the button, the tooltip no longer works because a disabled button fires absolutely no events. This means it doesn't fire focus or hover events, which are needed by the tooltip component. So instead of setting the `disabled` attribute, I decided to set the `aria-disabled` attribute, which **semantically** marks the element as disabled to assistive tech, and use the `canCreate` boolean in the `onClick` handler to disable the button's functionality, as well as hooking into the `aria-disabled` attribute in the styles for the button so that the button appears disabled to sighted users.

Reply:
Nice description of the approach here. Would it make sense to add a comment in the code for the next person to touch this about why `disabled` isn't being used?

Reply:
Yeah good idea! I added a comment.